Improve README

philipwastakenwastaken opened the pull request:
The current `README.md` is lacklustre. This PR seeks to improve the state of the readme.

- [x] Badges (build status, number of issues etc.)
- [x] Clarify that project contains both CLI and web app
- [x] Build guide
- [x] Documentation
- [x] Contribution guide
- [ ] Usage examples

Commit:
Added build status

BX326 committed:
Update README.md

The README has been extended with the following information:

* Mentions of the ``cli`` and ``web`` components
* Build instructions
* A short documentation section

BX326 commented:
The ``README.md`` now contains mentions of our different user interfaces as well as instructions for building the apps and how to generate documentation. I've added a bullet point regarding usage examples.

BX326 committed:
Create CONTRIBUTING.md

Short guidelines suggesting that contributors (1) write idiomatic code and (2) open descriptive pull requests.

BX326 committed:
Update README.md

Referring to the guidelines in CONTRIBUTING.md.

BX326 commented:
I've now added contribution guidelines in [``CONTRIBUTING.md``](https://github.com/url2ref/url2ref/blob/dba4879f82e173f5d102e34c3375bf6933192d51/CONTRIBUTING.md) according to [these GitHub instructions](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors). I've taken the liberty to suggest that contributors write idiomatic code and try to follow the existing style. The guidelines also instruct contributors to open descriptive pull requests.

BX326 commented:
We can merge whenever you approve these changes. I suggest we wait with usage examples until the application has been fledged out more.
